File: baekjoon/9863.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt';
const [...datas] = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split('\n')
  .map((x) => x.split(' ').map(Number));

const solution = (list) => {
  for (let i = 0; i < list.length; i++) {
    let [n, m, k] = list[i];
    if (!n && !m && !k) {
      return;
    }

    const caller = Array.from({ length: n }, (_, i) => i + 1);
    let cnt = 1;

    for (let j = 1; j <= k; j++) {
      while (cnt !== m) {
        cnt++;
        caller.push(caller.shift());
      }
      if (j === k) {
        console.log(caller[0]);
        break;
      }
      cnt = 1;
      caller.shift();
    }
  }
};

solution(datas);

File: baekjoon/9863.py
from collections import deque

while True:
    n, m, k = list(map(int, input().split()))
    if not n and not m and not k:
        break
    
    caller = deque([i for i in range(1, n + 1)])
    cnt = 1
    for i in range(k):
        while cnt != m:
            cnt += 1
            caller.append(caller.popleft())
        if i == k - 1:
            print(caller[0])
            break
        cnt = 1
        caller.popleft()